packages/flutter_tools/lib/src/web/bootstrap.dart

```diff
@@ -4,6 +4,320 @@
+  _ttPolicy = trustedTypes.createPolicy("flutter-tools-bootstrap", {
+    createScriptURL: (url) => {
+      let scriptUrl = _scriptUrls[url];
+      if (!scriptUrl) {
+        console.error("Unknown Flutter Web bootstrap resource!", url);
+      }
+      return scriptUrl;
+    }
+  });
+}
+
+// Creates a TrustedScriptURL for a given `scriptName`.
+// See `_scriptUrls` and `_ttPolicy` above.
+function getTTScriptUrl(scriptName) {
+  let defaultUrl = _scriptUrls[scriptName];
+  return _ttPolicy ? _ttPolicy.createScriptURL(scriptName) : defaultUrl;
+}
+
+(function() {
+  let appName = "$entrypoint";
+
+  // A uuid that identifies a subapp.
+  // Stubbed out since subapps aren't supported in Flutter.
+  let uuid = "00000000-0000-0000-0000-000000000000";
+
+  window.postMessage(
+      {type: "DDC_STATE_CHANGE", state: "initial_load", targetUuid: uuid}, "*");
+
+  // Load pre-requisite DDC scripts.
+  // We intentionally use invalid names to avoid namespace clashes.
+  let prerequisiteScripts = [
+    {
+      "src": getTTScriptUrl("moduleLoader"),
+      "id": "ddc_module_loader \x00"
+    },
+    {
+      "src": getTTScriptUrl("mapper"),
+      "id": "dart_stack_trace_mapper \x00"
+    }
+  ];
+
+  // Load ddc_module_loader.js to access DDC's module loader API.
+  let prerequisiteLoads = [];
+  for (let i = 0; i < prerequisiteScripts.length; i++) {
+    prerequisiteLoads.push(forceLoadModule(prerequisiteScripts[i].src));
+  }
+  Promise.all(prerequisiteLoads).then((_) => afterPrerequisiteLogic());
+
+  // Save the current script so we can access it in a closure.
+  var _currentScript = document.currentScript;
+
+  var afterPrerequisiteLogic = function() {
+    window.\$dartLoader.rootDirectories.push(_currentDirectory);
+    let scripts = [
+      {
+        "src": "dart_sdk.js",
+        "id": "dart_sdk"
+      },
+      {
+        "src": "main_module.bootstrap.js",
+        "id": "data-main"
+      }
+    ];
+    let loadConfig = new window.\$dartLoader.LoadConfiguration();
+    loadConfig.bootstrapScript = scripts[scripts.length - 1];
+
+    if (window.\$dartJITModules) {
+      loadConfig.loadScriptFn = function(loader) {
+        // Loads just the entrypoint module and required SDK modules.
+        let moduleSet = new Set();
+        // This cache is populated by ddc_module_loader.js
+        let libraryCache = JSON.parse(window.localStorage.getItem(`dartLibraryCache:\${appName}`));
+        if (libraryCache) {
+          // TODO(b/165021238) - when should this be invalidated?
+          moduleSet = new Set(libraryCache["modules"])
+        }
+        loader.addScriptsToQueue(scripts, function(script) {
+            // Preemptively load the module loader and previously executed modules.
+            return moduleSet.size == 0
+                  || script.id.includes("ddc_module_loader")
+                  // We preemptively load the stack_trace_mapper module so that we can
+                  // translate JS errors to Dart.
```

Comment:
```suggestion
            // Preemptively load the module loader and previously executed modules,
            // and the stack_trace_mapper so that we can translate JS errors to Dart.
            return moduleSet.size == 0
                  || script.id.includes("ddc_module_loader")
```

Reply:
(Not a fan of single-line comments in the middle of a multi-line conditional)

Reply:
Done!